Cabal/src/Distribution/Simple/GHC/Build/ExtraSources.hs

```diff
@@ -73,7 +74,15 @@ buildCSources
 buildCSources mbMainFile =
   buildExtraSources
     "C Sources"
-    Internal.componentCcGhcOptions
+    ( \verbosity lbi bi clbi odir filename ->
+        (Internal.sourcesGhcOptions verbosity lbi bi clbi odir filename)
+          { -- C++ compiler options: GHC >= 8.10 requires -optcxx, older requires -optc
+            -- we want to be able to support cxx-options and cc-options separately
+            ghcOptCxxOptions = Internal.separateGhcOptions (mkVersion [8, 10]) (compiler lbi) (Internal.defaultGhcOptCxxOptions lbi bi)
```

Comment:
This is probably silly but I'll ask anyway: why do we touch a cxx-related field in a function that's called `buildCSources`? Looks like only C-related options should be touched. 

I did read the comment above, which probably explains it, but I didn't understand how it's related to my conccern.

Reply:
hmm, I guess I should somehow refer to the tests `cabal-testsuite/PackageTests/FFI/ForeignOptsC`

```C
#include "clib.h"

#ifndef __TESTOPT_C__
#error "Did not get required __TESTOPT_C__ from cc-options"
#endif

#ifdef __TESTOPT_CXX__
#error "Got unexpected __TESTOPT_CXX__ from cxx-options"
#endif

int meaning_of_life_c() {
    return __TESTOPT_C__;
}
```
the test is just for this
```
executable foreign-opts-c-exe
  main-is:             Main.hs
  build-depends:       base
  default-language:    Haskell2010
  include-dirs:        cbits
  c-sources:           cbits/clib.c
  -- The following options are shared in all ForeignOpts cases:
  cc-options:          -D__TESTOPT_C__=11
  cxx-options:         -D__TESTOPT_CXX__=22
```